virtio-devices/src/block.rs

```diff
@@ -879,6 +887,35 @@ impl Block {
         self.writeback.store(writeback, Ordering::Release);
     }
 
+    pub fn resize(&mut self, new_size: u64) -> Result<()> {
+        if !new_size.is_multiple_of(SECTOR_SIZE) {
+            return Err(Error::InvalidSize);
+        }
+
+        match self.disk_image.resize(new_size) {
```

Comment:
please remove the whole nesting block by using:



```suggestion
let _ = self.disk_image.resize(new_size).map_err(Error::FailedDiskResize)?;
// the inner OK body ...
let nsectors = new_size / SECTOR_SIZE;     
// ...
```